### PR TITLE
Fix inconsistent work of date param in v2 analytics countries API and add it to v3

### PR DESCRIPTION
**Pull request by clearsense**

### CODE REVIEW

File: services/statsService.js [hunk 1]
```diff
@@ -57,38 +57,43 @@ ORDER BY
   try {
     const customStats = await getCustomStats();
 
-    const overriddenData = data.map(d => {
-      const customCountryStat = customStats.find(c => c.countryCode && d.countryCode && c.countryCode.toLowerCase() === d.countryCode.toLowerCase());
+    if(date) {
+      const overriddenData = data;
+    }
+    else {
+      const overriddenData = data.map(d => {
```
[the remark this answers is not in the document]
Comment (clearsense) [reply services/statsService.js]: I see, my bad. I will fix it now

[the remark this answers is not in the document]
Comment (clearsense) [reply services/statsService.js]: @xming13 is it okay now? I have add commit

**Comment by clearsense** [comment]
@xming13 hi, there was conflict, I fixed it. Can you check it, plz?

**Comment by xming13** [comment]
@clearsense please make pull request to dev branch

File: services/statsService.js [hunk 2]
```diff
@@ -57,38 +57,44 @@ ORDER BY
...
 
-    const overriddenData = data.map(d => {
-      const customCountryStat = customStats.find(c => c.countryCode && d.countryCode && c.countryCode.toLowerCase() === d.countryCode.toLowerCase());
+    const overriddenData = [];
+    if(date) {
+      overriddenData.push(...data);
+    }
+    else {
+      overriddenData.push(...data.map(d => {
+        const customCountryStat = customStats.find(c => c.countryCode && d.countryCode && c.countryCode.toLowerCase() === d.countryCode.toLowerCase());
```
Comment (xming13) [inline services/statsService.js]: you can write in this way
```
let overriddenData;
if (date) {
  overriddenData = data;
}
else {
  overriddenData = data.map(d => {
  ...
}
```

Comment (clearsense) [reply services/statsService.js]: @xming13 Yea, I feel that was bad code, thanks on corretion. But except that there is another question about this PR:
I see that /v3/analytics/country has been removed from API, but there is /v3/stats/worldometer/country method.

Can I add optional date param to last one mathod?

Comment (xming13) [reply services/statsService.js]: @clearsense yeah sure

**Comment by clearsense** [comment]
@xming13 done, but need testing

**Comment by clearsense** [comment]
@xming13 seems like I can't change PR branch to dev in existing PR, so I will made new one and you can reject that one

**Comment by clearsense** [comment]
I found how to do it, so new PR don't needed. Sorry, for bunch or comments, check this now, plz @xming13

File: routes/v3/analytics.js [hunk 3]
```diff
@@ -62,6 +62,15 @@ router.get('/trend/country', cache.route(), asyncHandler(async function(req, res
...
 
+  if (req.query.hasOwnProperty('date')) {
+    date = req.query.date
+
+    // enforce date format
+    if (moment(date, 'YYYY-MM-DD').format('YYYY-MM-DD') !== date) {
+      return res.json('Invalid date format. Date format should be YYYY-MM-DD')
+    }
+  }
+
```
Comment (xming13) [inline routes/v3/analytics.js]: Hi @clearsense this part can removed.

File: services/statsService.js [hunk 4]
```diff
@@ -57,38 +57,44 @@ ORDER BY
...
     const customStats = await getCustomStats();
 
-    const overriddenData = data.map(d => {
-      const customCountryStat = customStats.find(c => c.countryCode && d.countryCode && c.countryCode.toLowerCase() === d.countryCode.toLowerCase());
+    const overriddenData = [];
+    if(date) {
+      overriddenData = data;
+    }
+    else {
+      overriddenData = data.map(d => {
```
Comment (xming13) [inline services/statsService.js]: Hi @clearsense, `const` variable declaration cannot be reassigned, use `let` instead.

Comment (clearsense) [reply services/statsService.js]: Yeah, sure. Done

File: routes/v3/analytics.js (hunk 3 above)
Comment (xming13) [reply routes/v3/analytics.js]: @clearsense

Comment (clearsense) [reply routes/v3/analytics.js]: Missed that, fixed now @xming13

**Comment by clearsense** [comment]
@xming13 hi! Can you please tell me when this and #59 PR will be avalable on prod API?

**Pull request merged**